contracts/cw4-group/src/contract.rs

```diff
@@ -98,12 +101,12 @@ pub fn handle_update_members(
     // make the local update
     let diff = update_members(deps.branch(), env.block.height, info.sender, add, remove)?;
     // call all registered hooks
-    let mut ctx = Context::new();
-    for h in HOOKS.may_load(deps.storage)?.unwrap_or_default() {
-        let msg = diff.clone().into_cosmos_msg(h)?;
-        ctx.add_message(msg);
-    }
-    Ok(ctx.into())
+    let messages = prepare_hooks(deps.storage, |h| diff.clone().into_cosmos_msg(h))?;
```

Comment:
Clever. I guess `clone()` is cheap here. Plus. a new message must be created nevertheless. Plus, hooks are normaly a few.